proxy/logging/LogAccess.cc

```diff
@@ -643,6 +643,126 @@ LogAccess::unmarshal_str(char **buf, char *dest, int len, LogSlice *slice)
   return -1;
 }
 
+static int
+escape_json(char *dest, const char *buf, int len)
+{
+  ink_assert(buf != nullptr);
+
+  int escaped_len = 0;
+
+  for (int i = 0; i < len; i++) {
+    char c = buf[i];
+    switch (c) {
```

Comment:
Here are two alternatives to a switch statement:  https://godbolt.org/z/G3fz3469v

Reply:
Thanks for your suggestion.
I tried to modify `escape_json` using bitset but I found that I need the mapping like `\b` to `b`.

I made a modified version at
https://github.com/hnakamur/json-escape-cpp-experiment/blob/main/test.cpp

This version properly escapes the DEL `0x7f` character too.

Can I replace `escape_json` with this one?

Reply:
How about this?
```
namespace
{
class EscLookup
{
public:
  static const char NO_ESCAPE{'\0'};
  static const char LONG_ESCAPE{'\x01'};

  static char result(char c)
  {
    return _lu.table[static_cast<unsigned char>(c)];
  }

private:
  struct _LUT
  {
    _LUT();

    unsigned short table[0x7f];
  };

  inline static _LUT const _lu;
};

EscLookup::_LUT::_LUT()
{
  for (unsigned i = 0; i < ' '; ++i) {
    table[i] = LONG_ESCAPE;
  }
  for (unsigned i = '\x7f'; i < sizeof(table); ++i) {
    table[i] = LONG_ESCAPE;
  }

  // Short escapes.
  //
  table['\b'] = 'b';
  table['\t'] = 't';
  table['\n'] = 'n';
  table['\f'] = 'f';
  table['\r'] = 'r';
  table['\\'] = '\\';
  table['\"'] = '\\';
}

char nibble(int nib)
{
  return nib >= 0xa ? 'a' + (nib - 0xa) : '0' + nib;
}

} // end anonymous namespace

static int
escape_json(char *dest, const char *buf, int len)
{
  int escaped_len = 0;

  for (int i = 0; i < len; i++) {
    char c = buf[i];
    char ec = EscLookup::result(c);
    if (__builtin_expect(EscLookup::NO_ESCAPE == ec, 1)) {
      if (dest) {
        *dest++ = c;
      }
      escaped_len++;

    } else if (EscLookup::LONG_ESCAPE == ec) {
      if (dest) {
        *dest++ = '\\';
        *dest++ = 'u';
        *dest++ = '0';
        *dest++ = '0';
        *dest++ = nibble(static_cast<unsigned char>(c) >> 4);
        *dest++ = nibble(c & 0x0f);
      }
      escaped_len += 6;

    } else { // Short escape.
      if (dest) {
        *dest++ = '\\';
        *dest++ = ec;
      }
      escaped_len += 2;
    }
  } // end for
  return escaped_len;
}
```

Reply:
Thanks! I added commit https://github.com/apache/trafficserver/pull/8886/commits/ab9772927c23505b0b8b7e247428e0b5048364b4 before noticing your code.
I'll update it later.

Also I noticed slicing is done after escape now and produces incorrect strings like `\u0` or so.
I'll looking into it later this too.

Reply:
Copied the code at https://github.com/apache/trafficserver/pull/8886#discussion_r892896951 to https://github.com/apache/trafficserver/pull/8886/commits/a65b8b4b6284e049e860d85119ec311ce03a5ccf with specifying the author [ywkaras](https://github.com/ywkaras). Is it OK with you?

Reply:
> Copied the code at [#8886 (comment)](https://github.com/apache/trafficserver/pull/8886#discussion_r892896951) to [a65b8b4](https://github.com/apache/trafficserver/commit/a65b8b4b6284e049e860d85119ec311ce03a5ccf) with specifying the author [ywkaras](https://github.com/ywkaras). Is it OK with you?

Yes OK.